### PR TITLE
Adding Overloads up to T8.

### PR DESCRIPTION
**Pull request by mikaelm12**
Issue: https://github.com/aspnet/SignalR/issues/2626

### CODE REVIEW

File: clients/java/signalr/src/test/java/HubConnectionTest.java [hunk 1]
```diff
@@ -370,6 +370,139 @@ public void SendWithFiveParamsTriggersOnHandler() throws Exception {
...
+        assertEquals("B", value2.get());
+        assertEquals("C", value3.get());
+        assertTrue(value4.get());
+        assertEquals(12, value5.get(), 0);
+        assertEquals("D", value6.get());
+        assertEquals("E", value7.get());
+        assertEquals("F", value8.get());
+    }
+
+
```
Comment (jkotalik) [inline clients/java/signalr/src/test/java/HubConnectionTest.java]: nit

File: clients/java/signalr/src/test/java/HubConnectionTest.java [hunk 2]
```diff
@@ -370,6 +370,139 @@ public void SendWithFiveParamsTriggersOnHandler() throws Exception {
...
+        AtomicReference<String> value3 = new AtomicReference<>();
+        AtomicReference<Boolean> value4 = new AtomicReference<>();
+        AtomicReference<Double> value5 = new AtomicReference<>();
+        AtomicReference<String> value6 = new AtomicReference<>();
+        AtomicReference<String> value7 = new AtomicReference<>();
+
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("http://example.com", mockTransport);
+
+        hubConnection.on("inc", (param1, param2, param3, param4, param5, param6, param7) ->{
```
Comment (anurse) [inline clients/java/signalr/src/test/java/HubConnectionTest.java]: nit: space after `->`

File: clients/java/signalr/src/test/java/HubConnectionTest.java [hunk 3]
```diff
@@ -370,6 +370,139 @@ public void SendWithFiveParamsTriggersOnHandler() throws Exception {
...
+        AtomicReference<String> value2 = new AtomicReference<>();
+        AtomicReference<String> value3 = new AtomicReference<>();
+        AtomicReference<Boolean> value4 = new AtomicReference<>();
+        AtomicReference<Double> value5 = new AtomicReference<>();
+        AtomicReference<String> value6 = new AtomicReference<>();
+
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("http://example.com", mockTransport);
+
+        hubConnection.on("inc", (param1, param2, param3, param4, param5, param6) ->{
```
Comment (anurse) [inline clients/java/signalr/src/test/java/HubConnectionTest.java]: nit: space after `->`

File: clients/java/signalr/src/test/java/HubConnectionTest.java [hunk 4]
```diff
@@ -370,6 +370,139 @@ public void SendWithFiveParamsTriggersOnHandler() throws Exception {
...
+        AtomicReference<Boolean> value4 = new AtomicReference<>();
+        AtomicReference<Double> value5 = new AtomicReference<>();
+        AtomicReference<String> value6 = new AtomicReference<>();
+        AtomicReference<String> value7 = new AtomicReference<>();
+        AtomicReference<String> value8 = new AtomicReference<>();
+
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("http://example.com", mockTransport);
+
+        hubConnection.on("inc", (param1, param2, param3, param4, param5, param6, param7, param8) ->{
```
Comment (anurse) [inline clients/java/signalr/src/test/java/HubConnectionTest.java]: nit: space after `->`

**Comment by mikaelm12** [comment]
These new tests need to be updated to use the new testing transport.

**Pull request merged**